components/payments/cmd/api/internal/storage/transfer_initiation.go

```diff
@@ -105,6 +105,17 @@ func (s *Storage) transferInitiationQueryContext(qb query.Builder) (string, []an
 			default:
 				return "", nil, fmt.Errorf("unexpected type %T for column '%s'", accountID, key)
 			}
+		case key == "reference":
+			if operator != "$match" {
+				return "", nil, fmt.Errorf("'%s' column can only be used with $match", key)
+			}
+
+			switch reference := value.(type) {
+			case string:
+				return fmt.Sprintf("%s = ?", key), []any{reference}, nil
+			default:
+				return "", nil, fmt.Errorf("unexpected type %T for column '%s'", reference, key)
```

Comment:
```suggestion
				return "", nil, fmt.Errorf("unexpected type %T for column %q", reference, key)
```

you can also use `%q` to quote a string